diff --git a/node_modules/redux-little-router/es/components/fragment.js b/node_modules/redux-little-router/es/components/fragment.js
index 9f77e1b..018aba8 100644
--- a/node_modules/redux-little-router/es/components/fragment.js
+++ b/node_modules/redux-little-router/es/components/fragment.js
@@ -134,7 +134,8 @@ export var FragmentComponent = function (_Component2) {
           forNoMatch = _props.forNoMatch,
           location = _props.location,
           parentRoute = _props.parentRoute,
-          parentId = _props.parentId;
+          parentId = _props.parentId,
+          wrapper = _props.wrapper;
 
 
       var shouldShow = shouldShowFragment({
@@ -145,7 +146,7 @@ export var FragmentComponent = function (_Component2) {
       });
 
       if (!shouldShow && !forNoMatch) {
-        return null;
+        return wrapper ? wrapper(false, Children.only(children)) : null;
       }
 
       var currentRoute = resolveCurrentRoute(parentRoute, forRoute);
@@ -153,13 +154,13 @@ export var FragmentComponent = function (_Component2) {
       if (parentId) {
         var previousMatch = matchCache.get(parentId);
         if (previousMatch && previousMatch !== currentRoute) {
-          return null;
+          return wrapper ? wrapper(false, Children.only(children)) : null;
         } else {
           matchCache.add(parentId, currentRoute);
         }
       }
 
-      return Children.only(children);
+      return wrapper ? wrapper(true, Children.only(children)) : Children.only(children);
     }
   }]);
 
diff --git a/node_modules/redux-little-router/es/environment/browser-router.js b/node_modules/redux-little-router/es/environment/browser-router.js
index 5c7573f..339e331 100644
--- a/node_modules/redux-little-router/es/environment/browser-router.js
+++ b/node_modules/redux-little-router/es/environment/browser-router.js
@@ -1,6 +1,6 @@
 var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };
 
-import createBrowserHistory from 'history/createBrowserHistory';
+import {createBrowserHistory} from 'history';
 
 
 import normalizeHref from '../util/normalize-href';
diff --git a/node_modules/redux-little-router/es/environment/express-router.js b/node_modules/redux-little-router/es/environment/express-router.js
index 4194c59..767cd38 100644
--- a/node_modules/redux-little-router/es/environment/express-router.js
+++ b/node_modules/redux-little-router/es/environment/express-router.js
@@ -1,6 +1,6 @@
 
 
-import createMemoryHistory from 'history/createMemoryHistory';
+import {createMemoryHistory} from 'history';
 
 
 import normalizeHref from '../util/normalize-href';
diff --git a/node_modules/redux-little-router/es/environment/hapi-router.js b/node_modules/redux-little-router/es/environment/hapi-router.js
index f348297..9668fe9 100644
--- a/node_modules/redux-little-router/es/environment/hapi-router.js
+++ b/node_modules/redux-little-router/es/environment/hapi-router.js
@@ -1,6 +1,6 @@
 
 
-import createMemoryHistory from 'history/createMemoryHistory';
+import {createMemoryHistory} from 'history';
 
 
 import normalizeHref from '../util/normalize-href';
diff --git a/node_modules/redux-little-router/es/environment/hash-router.js b/node_modules/redux-little-router/es/environment/hash-router.js
index 33d4966..adeaefa 100644
--- a/node_modules/redux-little-router/es/environment/hash-router.js
+++ b/node_modules/redux-little-router/es/environment/hash-router.js
@@ -1,6 +1,6 @@
 var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };
 
-import createHashHistory from 'history/createHashHistory';
+import {createHashHistory} from 'history';
 
 
 import normalizeHref from '../util/normalize-href';
diff --git a/node_modules/redux-little-router/es/util/normalize-href.js b/node_modules/redux-little-router/es/util/normalize-href.js
index ab74aa0..379ccd8 100644
--- a/node_modules/redux-little-router/es/util/normalize-href.js
+++ b/node_modules/redux-little-router/es/util/normalize-href.js
@@ -2,7 +2,7 @@ var _extends = Object.assign || function (target) { for (var i = 1; i < argument
 
 function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }
 
-import { parsePath } from 'history/PathUtils';
+import { parsePath } from 'history';
 
 import qs from 'query-string';
 
